models/comments-model.js

```diff
@@ -0,0 +1,47 @@
+const db = require("../db/connection");
+const { selectArticles } = require("../models/articles-model");
+
+exports.selectComments = (articleId) => {
+  if (!Number.isInteger(Number(articleId))) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return db
+    .query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC",
+      [articleId]
+    )
+    .then((result) => {
+      const comments = result.rows;
+      if (comments.length === 0) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
+      }
+      return comments;
+    });
+};
+
+exports.createComments = (articleId, { username, body }) => {
+  if (
+    !Number.isInteger(Number(articleId)) ||
```

Comment:
You have made specific error handlers for codes in psql but are rejecting yourself. If a query is using the wrong types as defined by the seed psql will reject for you.

Reply:
<img width="761" alt="Screenshot 2024-08-27 at 14 14 14" src="https://github.com/user-attachments/assets/f448ecc1-26db-48a8-be38-1144628a8e44">